doc/lua_style_guide.md

```diff
@@ -397,7 +397,194 @@ If you have a simple build, we recommend using [LuaRock's built-in support for b
+
+### Constructor vs. Initializer
+
+In our chosen OOP style we distinguish between the constructor and the initializer.
+
+The constructor creates a new instance, the initializer initializes the instance fields with values.
+
+This distinction is not necessary in languages like C++ or Java.
+
+But in our case we need it to avoid multiple instantiation. We will see that in a bit.
+
+### Abstract Classes and Methods
+
+Concrete classes are classes that can be instantiated. Abstract classes on the other hand can't. Instead, they only serve as base classes from which other classes must be derived. 
+
+Abstract classes have an initializer, but no constructor.
+
+Abstract methods throw an error stating that the method is abstract.
+
+### Inheritance
+
+An elegant way to support inheritance is using [metatables][METATABLES]. If you haven't already, now is the perfect time to read up on that concept.
+
+Metatables allow you to define hook functions that take action when certain events happen on an object. For example when you try to concatenate them or turn them into a string.
+
+Or if you access a table attribute that does not exist. And that last trick is exactly what you need, if you want to emulate a [virtual method table](https://en.wikipedia.org/wiki/Virtual_method_table) like in native OO languages.
+
+You can find a full code example for inheritance under [examples/oop/inheritance.lua](../examples/oop/inheritance.lua).
+
+The example demonstrates classes that represent database objects.
+
+#### Implementing an Abstract Base Class
+
+Let's first create a base class. In our example it is an [abstract class](#abstract-classes-and-methods) because that keeps the example simpler and also serves to illustrate the difference between a concrete and an abstract class.
+
+First, we define the class as an empty table.
+
+```lua
+local AbstractDatabaseObject = {}
+```
+
+Next, we add an initializer that saves a parameter `name` in the instance variable (aka. "field") `name`.
+
+```lua
+function AbstractDatabaseObject:_init(name)
+    self.name = name
+end
+```
+
+You can see the `:` operator at work here again. `_init` operates on an instance and the hidden first parameter called `self` points to that instance.
+
+We also give the base class a getter for the name. Since all database objects have a name, we want that in the base class to avoid code duplication.
+
+```lua
+function AbstractDatabaseObject:get_name()
+    return self.name
+end
+```
+
+#### Deriving a Class From a Base Class
+
+Now, we derive a class `Table` from `AbstractDatabaseObject`.
+
+```lua
+local Table = {}
+Table.__index = Table
+setmetatable(Table, {__index = AbstractDatabaseObject})
+```
+
+The fist line creates the class as empty table. Next we give the class `Table` a field `__index`, that points to itself. While that looks ridiculous at first, it will make sense once you look at the constructor.
```

Comment:
```suggestion
The fist line creates the class as an empty table. Next we give the class `Table` a field `__index`, that points to itself. While that looks ridiculous at first, it will make sense once you look at the constructor.
```